### ISSUE TITLE
java.lang.NoClassDefFoundError: sun/misc/SharedSecrets for JDK11

### ISSUE THREAD

**Issue by cpdyj** [vert-x3/vertx-ignite#80]
I get this Exception:
```
java.lang.NoClassDefFoundError: sun/misc/SharedSecrets
```
Following the Ignite 's docs I add VM options:
```
--add-exports=java.base/jdk.internal.misc=ALL-UNNAMED
--add-exports=java.base/sun.nio.ch=ALL-UNNAMED
--add-exports=java.management/com.sun.jmx.mbeanserver=ALL-UNNAMED
--add-exports=jdk.internal.jvmstat/sun.jvmstat.monitor=ALL-UNNAMED
--add-exports=java.base/sun.reflect.generics.reflectiveObjects=ALL-UNNAMED
--illegal-access=permit
```
But it's not working. Should I add any options to `build.gradle` ? Thank you

**Comment by JohnMcWaggy** [comment]
You can work around this issue for now by updating the versions of the Apache Ignite packages that are consumed by your project. The issue seems to be that older versions of Ignite (2.3.0 used by this package) do not support Java9-11, while newer versions of Ignite do. For my project, I fixed this issue by adding the following to my POM (Maven). You could try a similar thing in Gradle. 

   ```
 <dependency>
      <groupId>org.apache.ignite</groupId>
      <artifactId>ignite-core</artifactId>
      <version>2.8.0</version>
    </dependency>
    <dependency>
      <groupId>org.apache.ignite</groupId>
      <artifactId>ignite-spring</artifactId>
      <version>2.8.0</version>
      <exclusions>
        <exclusion>
          <groupId>org.springframework</groupId>
          <artifactId>spring-aop</artifactId>
        </exclusion>
        <exclusion>
          <groupId>org.springframework</groupId>
          <artifactId>spring-tx</artifactId>
        </exclusion>
        <exclusion>
          <groupId>org.springframework</groupId>
          <artifactId>spring-jdbc</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

**Comment by tsegismont** [comment]
This will be fixed by upgrading to 2.8.0, see #82

**Comment by cpdyj** [comment]
Thanks for your reply. I will wait for update.

在 2020/4/23 16:27, Thomas Segismont 写道:
